elser/README.md

```diff
@@ -0,0 +1,133 @@
+## ELSER Track
+
+This track benchmarks compound weighted terms queries and BM25 match queries using OOTB settings,
+both sets of queries are run with the Block Max WAND optimisation disabled and enabled.
+Max WAND optimisation is controlled by the `track_total_hits` query parameter (set to `false` 
+to _enable_ Block Max WAND and `true` to _disable_).
+
+The `text_expansion` query rewrites to a compound weighted term query using the token weights
+retuned by evaluating the [ELSER](https://www.elastic.co/guide/en/machine-learning/master/ml-nlp-elser.html) model. For benchmarking, evaluating the NLP model is not necessary nor is it desired as the 
+purpose is to test the generated query. The query terms are pre-generated by the ELSER model
+and hard coded into the searches. 
+
+
+The BM25 match queries are present to establish a benchmark for comparrison with the
+weighted term queries. 
+
+
+### Generating the queries
+The weighted tokens are found by evaluating ELSER with the [Inference API](https://www.elastic.co/guide/en/elasticsearch/reference/current/infer-trained-model.html) passing the query text.
+The response of the inference call contains the weighted tokens which are converted to
+the query with the following `jq` command:
+
+```
+
+ jq '.inference_results | .[0] | .predicted_value | { "query": {"bool": { "should":  [ to_entries | .[] | {term: { "ml.tokens": { value: .key, boost: .value } } } ] } } } ' infer_output.json
+
+```
+In this example the file `infer_output.json` contains the response of the inference call.
+
+
+The benchmark considers queries with different numbers of terms. Again these queries are created from the inference 
+response, the following script creates the operations in `operations/weighted-token-queries.json`. The first argument to the script 
+is a file containing the inference response. 
+
+```
+#!/bin/sh
+
+if [ -z $1 ]
+then 
+	echo "Missing input file: usgae ./gen_queries.sh input_json"; 
```

Comment:
```suggestion
	echo "Missing input file: usage ./gen_queries.sh input_json"; 
```

Should we published the script in a separate file? There are other examples where the script (generally in python) is provided as a separate file.